File: Get Equal Substrings Within Budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int left = 0, right = 0, cost = 0, maxx = 0, n = s.size();
        vector<int> costs;

        while (right < n) {
            costs.push_back(abs((int) (s[right] - t[right])));
            cost += costs[right];

            while (cost > maxCost) {
                cost -= costs[left];
                left++;
            }

            right++;
            maxx = max(maxx, right-left);
        }
        return maxx;
    }
};

File: Get Equal Substrings Within Budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        left, right = 0, 0
        costs = []
        n = len(s)
        cost = 0
        maxx = 0
        while right < n:
            costs.append(abs(ord(s[right]) - ord(t[right])))
            cost += costs[right]

            while cost > maxCost:
                cost -= costs[left]
                left += 1

            right += 1
            maxx = max(maxx, right-left)
        return maxx
